docs/contributing/index.rst

```diff
@@ -4,104 +4,325 @@
+3. [Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
+the repository to have your own copy using a button at the top right corner,
+under your username  i.e:
+
+    https://github.com/<username>/geomstats
+
+4. Clone your forked repository using:
+
+    $ git clone git@github.com:<username>/geomstats.git
+
+5. It is recommended practice to add the main Geomstats repository as the upstream remote repository too:
+
+    $ cd geomstats
+    $ git remote add upstream git@github.com:geomstats/geomstats.git
+
+6. At this point you should have the Geomstats code on your macine ready for development.
+To continue development, create a new development branch:
+
+    $ git checkout -b feature-branch
+
+7. Verify tat you are on the new branch:
+
+    $ git branch
+    * feature-branch
+    master
+
+8. Verify remote configurattion:
+
+    $ git remote -v
+    upstream	https://github.com/geomstats/geomstats (fetch)
+    upstream	https://github.com/geomstats/geomstats (push)
+    origin	    https://github.com/<username>/geomstats (fetch)
+    origin	    https://github.com/<username>/geomstats (push)
+
+
+Dependencies and a virtual environment
+---------------------------------------
+
+We recommend using the conda virtual environment to create separation between your development environment
+from aany other Geomstats versions installed on your system. Create a virtual environment with conda using:
+
+$ conda create -n geomstats-dev
+
+Then clone your code to this environment and activate it:
+
+    $ git clone git@github.com:<username>/geomstats.git
+    $ cd geomstats
+    $ conda activate geomstats-dev
+
+Now install the Geomstats minimumcrequirements:
+
+    $ conda env create --file environment.yml
+
+.. note::
+   See the `setup.cfg` file for details on all project requirements
+
+.. _backends dev
+
+Backends
+--------
+
+Geomstats supports several backends, namely:
+
+- numpy
+- autograd
+- tensorflow
+- pytorch
+
+The default backend is `numpy, therefore install the other backends using:
+
+$ pip install geomstats[<backend_name>]
+
+And use an environment variable to set backed:
+
+$ export GEOMSTATS_BACKEND=<backend_name>
+
+Run the tests
+--------------
+
+Geomstats tests can be run in two ways, either using `nose` or one of the backends
+mentioned earlier.
+
+Using nose
+~~~~~~~~~~
+
+To run tests with Nose, first install the required packages:
+
+    $ pip install geomstats[dev]
+
+Then run all tests using:
+
+    $ nose2 tests
+
+You can run tests by file name:
+
+    $ nose2 tests.test__my_manifold
+
+Alternatively, run a particular test in a given test file:
+
+    $ nose2 tests.test__my_manifold.TestMyManifold.test_dimension
+
+Using a backend
+~~~~~~~~~~~~~~~~~
+
+We recommend that you run  all tests using the different backends discussed in 
+the `backends section <backends dev`_ . 
+
+Threfore install the backends as described before running all the tests using:
+
+    $ numpy tests
+
+Run a test file:
```

Comment:
I haven't done much with tests, so sorry if this is an ignorant comment, but do you mean "Test a single file"? From the sentence "Run a test file" i am not completely sure what is being done.

Reply:
There are some test files that you can run with pytest checking whether or not the associated scripts do not raise any error. For example if you run `test_hypersphere` with pytest it will test that `hypersphere` runs properly (that everything works if the test file is well-written) I hope this makes sense. This is explained later on I think but maybe it would make sense to put it before?